### PR TITLE
hex: Add from_spaced_hex()

### PR DESCRIPTION
**Pull request by chfast**
The `from_spaced_hex()` is `from_hex()` extension where the whitespace in the input is ignored.

### CODE REVIEW

File: include/evmc/filter_iterator.hpp [hunk 1]
```diff
@@ -32,10 +33,19 @@ template <typename BaseIterator,
           bool predicate(typename std::iterator_traits<BaseIterator>::value_type) noexcept>
 struct filter_iterator
 {
+    /// The iterators' difference type.
```
Comment (gumb0) [inline include/evmc/filter_iterator.hpp:36]: either `iterator's` or just `iterator`, make it consistent with comments below

File: include/evmc/filter_iterator.hpp [hunk 2]
```diff
@@ -54,26 +64,32 @@ struct filter_iterator
...
 
+    /// The increment operator.
     constexpr void operator++() noexcept
     {
         ++base;
         forward_to_next_value();
     }
 
-    constexpr bool operator!=(const filter_iterator& o) const noexcept { return base != o.base; }
+    /// The equal operator.
```
Comment (gumb0) [inline include/evmc/filter_iterator.hpp:88]: I think it should be "equality"/"inequality" operator 
https://docs.microsoft.com/en-us/cpp/cpp/equality-operators-equal-equal-and-exclpt-equal?view=msvc-170

File: include/evmc/filter_iterator.hpp (hunk 1 above)
Comment (chfast) [reply include/evmc/filter_iterator.hpp:36]: This was suppose to be "two iterators' difference type".

File: include/evmc/filter_iterator.hpp [hunk 3]
```diff
@@ -83,6 +99,7 @@ struct skip_space_iterator : filter_iterator<BaseIterator, is_not_space>
     using filter_iterator<BaseIterator, is_not_space>::filter_iterator;
 };
 
+/// CTAD.
```
Comment (gumb0) [inline include/evmc/filter_iterator.hpp:102]: a bit cryptic
```suggestion
/// Class template argument deduction guide.
```

File: include/evmc/filter_iterator.hpp (hunk 1 above)
Comment (gumb0) [reply include/evmc/filter_iterator.hpp:36]: Fine with changing to "Two"

**Review by gumb0 — APPROVED** [review]
Looks good, only minor suggestions in comments.

**Review by gumb0 — APPROVED** [review]

**Pull request merged**